src/anim/controller/anim-blend-tree-2d-directional.js

```diff
@@ -58,7 +58,7 @@ class AnimBlendTreeDirectional2D extends AnimBlendTree {
             const child = this._children[i];
             child.weight = child._weight / weightSum;
             if (this._syncAnimations) {
-                child.weightedSpeed = child.animTrack.duration / child.absoluteSpeed / weightedDurationSum;
+                child.weightedSpeed = child.animTrack.duration / child.absoluteSpeed / weightedDurationSum * weightSum;
```

Comment:
Is it just me or is the operator precedence a bit scary here? Anybody prefer some brackets?

Reply:
I was thinking the same: two divides... no brackets... whaaaa?! :D

Reply:
yeah. instead of `a / b / c` I prefer `a / (b * c)`

Reply:
I've split this onto two lines for a bit a clarity as to what's being calculated.

Reply:
Great. But I would still put the brackets around the first div in case readers forget that division takes precedence over multiplication.

Reply:
Done